app/src/main/java/com/example/sharingang/ItemsListFragment.kt

```diff
@@ -35,20 +36,31 @@ class ItemsListFragment : Fragment() {
         })
     }
 
-    override fun onCreateView(
-        inflater: LayoutInflater, container: ViewGroup?,
-        savedInstanceState: Bundle?
-    ): View {
-        val binding: FragmentItemsListBinding =
-            DataBindingUtil.inflate(inflater, R.layout.fragment_items_list, container, false)
-        binding.viewModel = viewModel
+    fun setupButtons() {
         binding.newItemButton.setOnClickListener { view: View ->
-            view.findNavController().navigate(R.id.action_itemsListFragment_to_newItemFragment)
+            view.findNavController().navigate(
+                ItemsListFragmentDirections.actionItemsListFragmentToNewItemFragment()
+            )
         }
         binding.goToMap.setOnClickListener { view: View ->
             goToMap(view)
         }
+        binding.userProfileButton.setOnClickListener {
+            val action =

```

Comment:
Why the need to seperate the action into a variable instead of doing what you did for newItemButton above?